packages/sdk/src/interfaces/text-to-image.ts

```diff
@@ -0,0 +1,81 @@
+import { InterfaceDeclaration } from '../integration/definition'
+import z from '../zui'
+
+const ImageModelRefSchema = z.object({
+  id: z.string(),
+})
+
+const ImageModelSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  costPerImage: z.number().describe('Cost per image generation, in U.S. dollars'),
+  sizes: z.array(z.string()).describe('Available image sizes'),
+  defaultSize: z.string().describe('Default image size generated by model'),
+})
+
+const ImageGenerationParamsSchema = z
+  .object({
+    id: z.string(), // TODO: remove
+  })
+  .describe('Model-specific parameters for image generation')
+
+const GenerateImageInputSchema = <TModelRef extends z.ZodSchema, TParams extends z.ZodSchema>(
+  imageModelRefSchema: TModelRef,
+  paramsSchema: TParams
+) =>
+  z.object({
+    model: imageModelRefSchema.optional().describe('Model to use for image generation'),
+    prompt: z.string(),
+    size: z.string().optional(),
+    params: paramsSchema.optional(),
+  })
+
+const GenerateContentInputBaseSchema = GenerateImageInputSchema(ImageModelRefSchema, ImageGenerationParamsSchema)
+
+const GenerateImageOutputSchema = z.object({
+  model: z.string().describe('Model name used'),
+  imageUrl: z.string().describe('Temporary URL of generated image'),
+  cost: z.number().describe('Cost of the image generation, in U.S. dollars'),
```

Comment:
**note**: I'll create a PR for the bridge to charge the image generation cost as AI Spend to the workspace of the bot that requested this action.